### PR TITLE
fix: Correctly initialize value of GeoCoordinate.Point on Android

### PR DESCRIPTION
**Pull request by jcantin-nventive**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

When using Geolocator to obtain a Geocoordinate position, Geocoordinate.Latitude and Geocoordinate.Longitude are correct but Geocoordinate.Point.Latitude and Geocoordinate.Point.Longitude (which are the correct way to obtain those values according to MSDN doc) are all set to 0, because we use a ctor which is not implemented.

## What is the new behavior?

Geocoordinate.Point is correctly set.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [ ] ~~[Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)~~ (I have tested that the values are set correctly)
- [ ] ~~Validated PR `Screenshots Compare Test Run` results.~~
- [x] Contains **NO** breaking changes
- [ ] ~~Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).~~
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs [hunk 1]
```diff
@@ -125,9 +125,7 @@ public static Geoposition ToGeoPosition(this Location location)
 							Latitude = location.Latitude,
 							Longitude = location.Longitude,
 							Altitude = location.Altitude,
-						},
-						AltitudeReferenceSystem.Ellipsoid,
-						Wgs84SpatialReferenceId
+						}
```
Comment (MatFillion) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:128]: constant Wgs84SpatialReferenceId was used for which reason? 

Also I suppose you can remove the private constant as it's not used anymore.

Comment (jcantin-nventive) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:128]: I don't know why it was added, but all it did was break the code.

File: src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs [hunk 2]
```diff
@@ -125,9 +123,7 @@ public static Geoposition ToGeoPosition(this Location location)
 							Latitude = location.Latitude,
 							Longitude = location.Longitude,
 							Altitude = location.Altitude,
-						},
-						AltitudeReferenceSystem.Ellipsoid,
-						Wgs84SpatialReferenceId
```
Comment (jeanplevesque) [inline src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: By removing this, are we losing some functionality (that we may not understand right now)?
Wouldn't it be better to fix the Geopoint ctor instead?

Comment (jcantin-nventive) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: We never had this functionality. It was added for wasm only by this commit. https://github.com/unoplatform/uno/commit/b7bba3d68e3ef597bedb8ebd892d2e4f14b059b5#diff-60dce75981e870b5855507ba793dbf87

These properties don't exist in UWP.

Comment (jeanplevesque) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: But if this was added for WASM, are we breaking something by removing it?

Comment (pkar70) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: AltitudeReferenceSystem.Ellipsoid and Wgs84SpatialReferenceId is also in #2536.

Comment (davidjohnoliver) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: From the history, it looks like some changes that are relevant to WASM, were mistakenly also applied to Android. Maybe @MartinZikmund can confirm but this fix looks good to me.

Comment (pkar70) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: Correct (full) ToGeoPosition() method SHOULD put AltitudeReferenceSystem.Ellipsoid and Wgs84SpatialReferenceId values. Current implementation (at least, Android version) should be corrected, as in my PR #2536. Also, some other changes should be done, but this PR (2536) has small chances to be merged.

Comment (davidjohnoliver) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs:130]: @pkar70 thanks for the clarification, on closer inspection it looks like you're right, the better fix is to make the correct ctor available for Android.

Comment (jcantin-nventive) [reply src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs]: @pkar70 @davidjohnoliver OK, the changes we discussed have been made!

**Comment by MartinZikmund** [comment]
@davidjohnoliver @jcantin-nventive I added the hardcoded AltitudeReferenceSystem because the Android docs state that this is always used (see https://gis.stackexchange.com/a/75412)

**Pull request merged**